src/cls/rbd/cls_rbd.cc

```diff
@@ -3130,7 +3130,7 @@ int dir_remove_image(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
  */
 int dir_state_assert(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
 {
-  cls::rbd::DirectoryState directory_state;
+  cls::rbd::DirectoryState directory_state = -1;
```

Comment:
Odd that the compiler complains about this since it can see the implementation of ```decode``` -- but why not actually set it to a real enum value since it's guaranteed to be overridden (or the function exits w/ an error)?